### ISSUE TITLE
Question about using model on my datasets

### ISSUE THREAD

**Issue by keruoya** [XiudingCai/EnCo-pytorch#1]
Hi!Thank you for your excellent work!I've noticed some strange results after training my dataset with your model for 200 epochs. The fake_B effect isn't very good, and cam_fake isn't clear either. What can I do? Should I adjust parameters or...?

**Comment by XiudingCai** [comment]
Hi there,


Thank you for your interest in our work.
&nbsp;To address the concerns regarding the `fake_B` effect and the clarity of `cam_fake`, here are a few suggestions that you might find useful:


1. **Adjusting Feature Constraints**: For tasks involving significant deformation, consider reducing the constraints on shallow-level features (e.g., removing constraints like `<3,28&gt;`). Conversely, if the layout of the generation is more critical, increasing the constraints on shallow-level feature pairs can be beneficial (e.g., adding constraints like `<0,31&gt;`).


2. **Experimenting with Different Loss Functions**: You could try out both the Least Squares GAN (LSGAN) and Hinge GAN loss functions, which are both implemented in our codebase. Each loss function has its own merits and might yield different results depending on your specific dataset and task.


3. **Loss Curve Analysis**: Providing us with a plot of the loss curves during the training process could be extremely helpful. This visual data can offer insights into what might be going wrong and how we can assist you in troubleshooting.


Please feel free to share the loss curves or any other relevant information, and we'll be more than happy to take a closer look and offer more targeted advice.


Best regards! :&gt;



---Original---
From: ***@***.***&gt;
Date: Thu, May 2, 2024 09:37 AM
To: ***@***.***&gt;;
Cc: ***@***.***&gt;;
Subject: [XiudingCai/EnCo-pytorch] Question about using model on my datasets(Issue #1)




 
Hi!Thank you for your excellent work!I've noticed some strange results after training my dataset with your model for 200 epochs. The fake_B effect isn't very good, and cam_fake isn't clear either. What can I do? Should I adjust parameters or...?

**Comment by keruoya** [comment]
您好！非常感谢您的回复，非常荣幸！这是loss文件，但是我不清楚要如何让他生成曲线，然后这是200代生成的fake_B以及对应的real_B，然后cam_fake一直是这样模糊的样子，请问这样是正常的吗？
fake_B
![5{J}@BUD`ZA`D8`R~7LMVOW](https://github.com/XiudingCai/EnCo-pytorch/assets/85153082/11f2d0c6-fa02-4adb-b2eb-31271f8f922a)


real_B
![(6FAT)M)I)NXB(HW6FR($DQ](https://github.com/XiudingCai/EnCo-pytorch/assets/85153082/f83a9a00-07ae-4da2-9aa5-68bb96ea71f2)


fake_cam
![ON$_`T)1NZ%7H~39WK2L Q1](https://github.com/XiudingCai/EnCo-pytorch/assets/85153082/ae2b8682-a05b-43ff-baf8-d51b10d6c20f)

损失文件
[loss_log.txt](https://github.com/XiudingCai/EnCo-pytorch/files/15183254/loss_log.txt)

**Comment by XiudingCai** [comment]
抱歉回复晚了。1. 您可以在训练前，通过命令行开启可视化脚本：python-m visdom.server，这样您可以在浏览器观察训练过程各个loss的变化，以更方便找到问题。2. cam图反映了生成图像的真假性，帮助patch的采样。从您提供的图片来看，cam图是正常的。3. 从fake B生成的结果来看，并不理想，这可能和生成任务难度有关。不知道你们的数据量是多少，数据预处理、数据数量、训练次数，都对生成结果质量有影响。



---Original---
From: ***@***.***&gt;
Date: Thu, May 2, 2024 10:47 AM
To: ***@***.***&gt;;
Cc: "Xiuding ***@***.******@***.***&gt;;
Subject: Re: [XiudingCai/EnCo-pytorch] Question about using model on mydatasets (Issue #1)




 
您好！非常感谢您的回复，非常荣幸！这是loss文件，但是我不清楚要如何让他生成曲线，然后这是200代生成的fake_B以及对应的real_B，然后cam_fake一直是这样模糊的样子，请问这样是正常的吗？
 fake_B
 ***@***.*** (view on web)
 
real_B
 6FAT.M.I.NXB.HW6FR.DQ.png (view on web)
 
fake_cam
 ON._.T.1NZ.7H.39WK2L.Q1.png (view on web)
 
损失文件
 loss_log.txt

**Comment by XiudingCai** [comment]
Hi,

After checking, we found no issues here, so we're closing this. If anything comes up, please feel free to reopen!

Thanks!
